spec/formatting.md

```diff
@@ -18,3 +18,71 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is invalid.
+- **Resolution errors** occur when the runtime value of a part of a message
+  cannot be determined.
+
+  - **Unresolved Variable errors** occur when a variable reference cannot be resolved.
+
+- **Selection errors** occur when message selection fails.
+
+  - **Selector errors** are failures in the matching of a key to a specific selector.
+  - **Missing Fallback errors** occur when no Variant is selected
```

Comment:
Sure, but what if we still end up here? If the source syntax isn't MF2, it's theoretically possible that we end up at runtime with a message data model which does not include an ultimate fallback/default variant. If that happens, it would be incorrect of us to claim this as bad syntax; hence categorising it here as a Selection Error.

Reply:
This is now a Data Model error.